Rubberduck.Core/UI/UnitTesting/TestExplorerViewModel.cs

```diff
@@ -143,6 +149,50 @@ public bool ExpandedState
             }
         }
 
+        public System.Collections.ObjectModel.ObservableCollection<string> OutcomeFilters { get; }
+
+        private string _testNameFilter = string.Empty;
+        public string TestNameFilter
+        {
+            get => _testNameFilter;
+            set
+            {
+                if (_testNameFilter != value)
+                {
+                    _testNameFilter = value;
+                    OnPropertyChanged();
+                    Tests.Filter = item => FilterResults(item);
+                }
+            }
+        }
+
+        private static readonly string _allResultsFilter = TestExplorer.ResourceManager.GetString("TestExplorer_AllResults", CultureInfo.CurrentUICulture);
+        private string _selectedOutcomeFilter = _allResultsFilter;
+        public string SelectedOutcomeFilter
+        {
+            get => _selectedOutcomeFilter;
+            set
+            {
+                if (_selectedOutcomeFilter != value)
+                {
+                    _selectedOutcomeFilter = value.Replace(" ", string.Empty);
+                    OnPropertyChanged();
+                    Tests.Filter = item => FilterResults(item);
+                }
+            }
+        }
+
+        private bool FilterResults(object unitTest)
+        {
+            OnPropertyChanged(nameof(Tests));
```

Comment:
this should not be the first thing that happens.

Reply:
Is the resolution for this to move `OnPropertyChanged` to just before the `return` statement?

Reply:
yes, it is :)